crypto/encryptmegolm.go

```diff
@@ -228,13 +228,21 @@ func (mach *OlmMachine) ShareGroupSession(ctx context.Context, roomID id.RoomID,
 	}
 
 	if len(fetchKeys) > 0 {
-		log.Debug().Strs("users", strishArray(fetchKeys)).Msg("Fetching missing keys")
-		for userID, devices := range mach.fetchKeys(ctx, fetchKeys, "", true) {
-			log.Debug().
-				Int("device_count", len(devices)).
-				Str("target_user_id", userID.String()).
-				Msg("Got device keys for user")
-			missingSessions[userID] = devices
+		if mach.DisableKeyFetching {
+			log.Warn().Strs("users", strishArray(fetchKeys)).Msg("Keys missing but key fetching is disabled")
```

Comment:
Sending requires network requests by definition so no point in disabling extra requests here